meson.build

```diff
@@ -117,8 +117,8 @@ add_global_link_arguments(
   language: 'c'
 )
 
-# ONLY for CI, do not enable this for distro builds
-if get_option('enable-werror')
+# ONLY for CI and only when using gcc, do not enable this for distro builds
+if get_option('enable-werror') and meson.get_compiler('c').get_id() == 'gcc'
```

Comment:
Does clang not support -Werror?

Reply:
It supports `-Werror`, but doesn't support the `-Wnoerror=cpp`

Reply:
Ahh, okay.